### DISCUSSION TITLE
Retrieving an Auth "Refresh" Token

### DISCUSSION THREAD

**Discussion by kevinfiol** [lighttube-org/LightTube#117] [Q&A] [answered]
Hi, great project!

I've tried following the guide here: https://github.com/lighttube-org/InnerTube/wiki/Authorization#using-a-refresh-token

However, when following the steps and logging into YouTube TV, I'm unable to see any Network Requests to `https://www.youtube.com/o/oauth2/token`. I made sure to have my adblock turned off while doing this.

My best guess is that the request name has changed. Looking to see if I can get some clarity here. Thanks.

**Comment by kuylar** [comment, marked as answer]
Just checked, and the requests appear on my side. Though I have found out that the user agent in the wiki does not work anymore. Can you try with this one? `Mozilla/5.0 (PlayStation; PlayStation 4/8.50) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/12.0 Safari/605.1.15`

If you still cannot see the requests, after a successful log in, you should try runing these scripts in the javascript console. The output should be the value you need

```js
JSON.parse(localStorage.getItem("yt.leanback.default::last-identity-used")).data.identityCredentials.refreshToken
```

```js
var data = JSON.parse(localStorage.getItem("yt.leanback.default::available-identities")).data;data[Object.keys(data)[0]].identityCredentials.refreshToken
```

(both of them basically get the same data from different localStorage entries, I recommend trying the first one first)

**Reply by kevinfiol** [reply]
Thanks for the prompt reply.

I used the user agent above, and after a successful login, looked in the Network tab. However, I still did not find a network request for `/o/oauth` there. Looked through this list and tried filtering for `oauth`.

However, grabbing the key from localStorage worked a treat. Thanks for your help.

![image](https://github.com/lighttube-org/LightTube/assets/4752973/ef639749-bc7d-4599-b8b6-aa934a1fe676)

**Reply by kuylar** [reply]
> after a successful login, looked in the Network tab
Looks like that's what you did wrong. The devtools has to be open during the sign in process. 

Thanks for reporting though, I'll be adding these details to the wiki.
